spec/factories/reference_requests.rb

```diff
@@ -0,0 +1,34 @@
+FactoryBot.define do
+  factory :reference_request do
+    referee
+    token { SecureRandom.uuid }
+    status { :created }
+
+    after(:stub) do |request|
+      request.email = request.referee&.email
+    end
+
+    after(:build) do |request|
+      request.email = request.referee&.email
+    end
+  end
+
+  trait :reference_requested do
+    status { :requested }
+
+    after(:create) do |request|
+      request.created!
+      request.requested!
```

Comment:
Question: Why are we doing this chain of status update writes in DB? These calls are so explicit that there must be a need for it, but I am completely missing it 😅 

I see the downside of the multiple DB writes that will happen each time one of these factories is created. But there must be an upside here! 🤔

Reply:
This is a way to create the data used by the timeline component.
That trigger the model callback that papertrail adds to the model to create differents `versions` entries

Reply:
IMHO the create should just be a create - there might be a trait which calls requested! (the created! call appears to be unnecessary, as that is the default state on line 5). Otherwise the tests result in different data from the code flows

Reply:
:recycle: